server/http.go

```diff
@@ -509,31 +449,24 @@ func (p *Plugin) deauthorizeUser(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	rawInfo, appErr := p.API.KVGet(zoomTokenKeyByZoomID + req.Payload.UserID)
-	if appErr != nil {
-		http.Error(w, appErr.Error(), http.StatusInternalServerError)
-		return
-	}
-
-	var info ZoomUserInfo
-	err := json.Unmarshal(rawInfo, &info)
+	info, err := p.fetchOAuthUserInfo(zoomTokenKeyByZoomID, req.Payload.UserID)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	if err = p.disconnect(info.UserID); err != nil {
+	if err = p.disconnectOAuthUser(info.UserID); err != nil {
 		http.Error(w, "Unable to disconnect user from Zoom", http.StatusInternalServerError)
 		return
 	}
 
-	err = p.dm(info.UserID, "We have received a deauthorization message from Zoom for your account. We have removed all your Zoom related information from our systems. Please, connect again to Zoom to keep using it.")
-	if err != nil {
+	message := "We have received a deauthorization message from Zoom for your account. We have removed all your Zoom related information from our systems. Please, connect again to Zoom to keep using it."
+	if err = p.sendDirectMessage(info.UserID, message); err != nil {
 		p.API.LogWarn("failed to dm user about deauthorization", "error", err.Error())
 	}
 
-	if req.Payload.UserDataRetention == "true" {
-		if err := p.zoomClient.CompleteCompliance(req.Payload); err != nil {
+	if req.Payload.UserDataRetention == "false" {
```

Comment:
Here I attempted to fix what appears to be a bug according to this: https://marketplace.zoom.us/docs/api-reference/data-compliance
> If the value of `user_data_retention` is TRUE, it means that the user has granted permission to retain their data beyond the ten days period stated in the Zoom App Marketplace Developer Agreement. In this case, the Data Compliance API does not need to be called.